File: app/src/main/java/com/test/upcoming/networkcall/WebService.java
package com.test.upcoming.networkcall;

import com.google.gson.JsonObject;
import com.test.upcoming.model.MovieImages;
import com.test.upcoming.model.Movies;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Observable;

public interface WebService {

    @GET("upcoming")
    Single<Movies> getMovieList(@Query("api_key") String id);


    @GET("{id}/images")
    Single<MovieImages> getMovieImages(@Path("id") String id, @Query("api_key") String apikey);

}

File: app/src/main/java/com/test/upcoming/networkcall/WebService.kt
package com.test.upcoming.networkcall

import com.google.gson.JsonObject

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import rx.Observable

interface WebService {

    @GET("upcoming")
    fun getMovieList(@Query("api_key") id: String): Observable<JsonObject>


    @GET("{id}/images")
    fun getMovieImages(@Path("id") id: String, @Query("api_key") apikey: String): Observable<JsonObject>

}
